### PR TITLE
Add StoryBook

### LINKED ISSUE #252: Add Storybook
https://storybook.js.org/

### PR DESCRIPTION
**Pull request by Vadorequest**
# Status

**STABLE**, the biggest issues have been fixed (babel/webpack/Emotion 10-11/Node vs browser/React providers).

~~Currently studying Storybook addons (first time using SB) to see what should be included by default.~~

~~SB is now configured with proper plugins, adding more stories (ideally, all components should have their stories)~~.

All NRN components have been documented thoroughly.

---

Resources:
- https://storybook.js.org/docs/react/get-started/install
- https://dev.to/cuichenli/how-do-i-setup-my-nextjs-development-environment-2kao
- https://github.com/cuichenli/nextjs-typescript-cypress-storybook-tailwind

# Issues

- ⚠️ https://github.com/emotion-js/emotion/issues/1606#issuecomment-758065127 `css` prop isn't recognized in some part of the code  
  - It doesn't seem to be a compilation issue (Next.js compiles properly locally and on Vercel). The error is only visible on the WebStorm IDE. Maybe a TS type override might fix the issue, until Storybook upgrades to Emotion 11? (couldn't find a way to override TS types `DOMAttributes` in [`node_modules/@emotion/core/types/index.d.ts`](https://gist.github.com/Vadorequest/effaaacdb9630504cae0c1835e2b36ae#file-emotion-core-types-index-d-ts-L82-L86))
  - To solve this, I've `@ts-ignore` the few statements where errors were thrown (~6)
- ✅ Emotion 11 requires to use a custom babel file. I first tried using the same as the one used by Next.js (`babel.config.js`) but it uses `automatic` runtime, which isn't supported by MDX, which is used in Storybook. So, I had to create a custom `.storybook/babel.config.js` file using `classic` mode. See https://emotion.sh/docs/css-prop#babel-preset
- ✅ NRN defines many providers, those need to be mocked using `decorators` in `.storybook/preview.js`
- ✅ Next.js `router` needs to be mocked (decorator), I used `storybook-addon-next-router`
- ✅ Sentry uses `@sentry/node`, webpack was configured to replace it by `@sentry/browser` during compilation
- ✅ The Airtable dataset needed to be mocked
- ✅ Various issues with node-related modules `fs`, etc. (webpack config)
- ✅ Added support for CSS Modules, used `storybook-css-modules-preset` (they're supported natively in Next.js) - See https://github.com/storybookjs/storybook/issues/9610#issuecomment-758550696 and https://github.com/vercel/next.js/issues/19345

# Features

## Storybook static site hosted on Vercel

- Upon code change, automatically export the Storybook static site and deploys it on Vercel

This way, you always have an online up-to-date version of your components!

Demo: https://nrn-v2-mst-aptd-at-lcz-sty-storybook.vercel.app/ (auto-updated)

## Support for MDX, TSX, Emotion 11 and CSS Modules

- Our components can be used normally, with TypeScript and Emotion or CSS Modules _(that wasn't as easy to implement as one might believe)_

## I18n support in Stories

- Our stories can be localized using a custom `locale` in the Storybook top toolbar! 
- Locize translations are being fetched upon build and then used within stories (using a filesystem cache to avoid fetching Locize API during navigation, to reduce cost).

---

# Usage

## Storybook localhost

Run `yarn sb`

## Storybook online

Automatically built + deployed on another Vercel project (dedicated project), as a static site.

https://nrn-v2-mst-aptd-at-lcz-sty-storybook.vercel.app/

Looking for a way to optionally protect the static site behind Basic Auth.

### CODE REVIEW

**Comment by jrmcj** [comment]
This PR was included in a batch that successfully built, but then failed to merge into master (it was a non-fast-forward update). It will be automatically retried.

**Comment by jrmcj** [comment]
> ✅ This PR was included in a batch that successfully built, but then failed to merge into master (it was a non-fast-forward update). It will be automatically retried.

**Comment by Vadorequest** [comment]
@jrmcj What do you mean?

File: .github/workflows/deploy-vercel-production.yml [hunk 1]
```diff
@@ -71,6 +70,7 @@ jobs:
         # Workflow overview:
         #   - Resolve customer to deploy from github event input (falls back to resolving it from vercel.json file)
         #   - Deploy the customer in production
+        #   - Creates multiple deployment aliases based on the "alias" property defined in the vercel.json file, and link them to the deployment
         # XXX You can use https://jqplay.org/ if you want to play around with "jq" to manipulate JSON
         run: |
```
Comment (jrmcj) [inline .github/workflows/deploy-vercel-production.yml:75]: J767!!'

**Review by jrmcj — CHANGES_REQUESTED** [review]

**Pull request merged**